### ISSUE TITLE
Ads are showing again

### ISSUE THREAD

**Issue by ghzgod** [yuliskov/SmartYouTubeTV#597]
Been using this apk for about a year and some change on my Fire sticks at home.  I noticed ads are now showing in videos where previously they never showed up.  I figure Google has changed their code to circumvent apps like this one from blocking  ads.  Just wanted to provide a heads up.

**Comment by foreveryoungforty** [comment]
Hello,
I think it's best you provide some examples (YT-links) and sreenshots of ads, that's easier for development to search for a solution.
Thanks.

**Comment by dinki** [comment]
I've started seeing ads as well.  I'm running stable in my living room and beta in bedroom.  Both started showing ads a few days ago.

Here are two YT links that played ads for me today on stable:
https://www.youtube.com/watch?v=UOLScQNdHus
https://www.youtube.com/watch?v=-MfuHyBoJGE

These are standard full screen ads with the 'skip this ad' button or the 'you can skip ad after x seconds' pop ups.

I'd like to help troubleshoot more if I can.

**Comment by foreveryoungforty** [comment]
Strange those video links do not give ads here in Belgium not on stable 697, not on beta 701....

**Comment by dinki** [comment]
Also seeing them in the beta.  Sorry no more links.  I did try clearing cache and data and a few days ago on beta but still the same.  Odd thing is sometimes when I down arrow to go to the 'more videos like this' or whatever it's called once you down arrow past the gui controls, I see a popup that says 'you can skip this ad after 1 sec' but it doesn't go away until I hit back button.

Not sure what's going on but YT has definitely changed something I'd guess.

The stable is on a Mibox 3 and the beta is on Nexus Player.

**Comment by yuliskov** [comment]
Anyone, who experienced problem with banner, please send the log on the
latest beta.

Regards, Yuriy

**Comment by dinki** [comment]
Will do.  Forgive me but can you tell me how to access the log?

**Comment by yuliskov** [comment]
Press Back in the main loading screen.
Then open Settings > Save log to file

Regards, Yuriy

**Comment by dinki** [comment]
I wasn't able to get a full ad to play, but I did get it to pop up the screen I was talking about:
https://i.imgur.com/ZQJVJPZ.png
This stays on the screen until you hit back.  The spinning circle spins as well.

Here's the logs if this helps any:
https://www.dropbox.com/s/fu511n7a8hstd26/log.zip?dl=0

**Comment by dinki** [comment]
@yuliskov I did a bit more testing.  First, I found that I had Smart Youtube and the Beta installed at the same time on both devices.  I thought it was odd that was the case as I thought the beta just overwrote Smart YT.  Thinking that might be the problem I uninstalled Smart YT, cleared cache/data on the beta.  I am not seeing ads on the beta and to be honest I'm not sure if I ever did.  I know I did see them on stable though.  What I am experiencing, however, is what I mentioned and took a photo of above.

I am unable to press down and go the suggestions.  This is surprisingly very problematic for me as I've come to find out that I use that functionality a lot.  I've discovered this when I was unable to use that functionality.  There is no way I can find to get that 'ad skip after x seconds' off the screen short of hitting back and exiting the playing video.

Do you have any thoughts on this?  I can certainly provide more logs if needed but I'm guessing they won't show anything different than what I've provide above.

Thanks

**Comment by dinki** [comment]
@yuliskov 
Me again.  So I tried to completely uninstall (clear data/cache then uninstall) both Smart YT and Beta apps then reinstalled the Beta after rebooting.  Still happening.  Strange thing is that my settings (like skip adverts) were still set when I launched the beta after what I thought would be a very clean install.  Is there a directory with the settings that is not being wiped when I uninstall?  Maybe it's something weird with those settings that's causing this problem?

**Comment by foreveryoungforty** [comment]
@dinki When updating sytv and starting the updated app, on my android tv there is a data folder on the main explorer level that has the settings saved of the previous install.
If you delete and do a fresh install, please be sure that that folder is no longer there.

**Comment by dinki** [comment]
@foreveryoungforty Thanks for the tip.  I did go back and cleared everything out including that directory.  I'm not certain that the settings were a holdover or not.  Is skip adverts on by default?

At any rate, I am unable to get past the problem I've described and photographed above.  I have the same behavior happening on two separate and different devices.  Is this issue isolated only to me?

**Comment by foreveryoungforty** [comment]
@dinki    please be sure to check the settings of the app (holding back button when starting up) that blocking ads is enabled. Cleaning the cache of the app might help.
I have no ads at all since your post, not on smart 697, not on beta 702.
Ads can be the result of your localization.

**Comment by dinki** [comment]
@foreveryoungforty Thanks for the information.  I have only started seeing these adverts within the past two weeks.  As stated above, I have cleared cache/data, removed directories, rebooted before reinstall etc and this persists.

A question about logging: will any of this show up in the logs?  What would I see in the logs?  Know that these popups show up even while the video is playing.  This shows up any time that I press down button more than twice.

**Comment by foreveryoungforty** [comment]
@dinki   I not an expert in reading the loggings ; I do translation, beta testing and of course I use the app on daily basis :-) 
A few months ago i had some large ad on the home screen, and sent the log.txt to the developer.

**Comment by dinki** [comment]
@foreveryoungforty Understood and thanks again.  I have posted a log above.  Perhaps I'm just impatient :)

Just took a look at the logs and seeing a BUNCH of stuff like this:

...
ERROR: TypeAdapter: VideoItem: Missing property in path $['longBylineText']['runs'][0]['navigationEndpoint']
ERROR: TypeAdapter: VideoItem: Missing property in path $['longBylineText']['runs'][0]['navigationEndpoint']
ERROR: TypeAdapter: VideoItem: Missing property in path $['lengthText']['runs']
...
Tons of others that refer to TypeAdapter:VideoItem

Is that normal?

You can probably tell by my persistence and posts that I'd really like to get this thing working well again.  Thank you all for the guidance.

**Comment by foreveryoungforty** [comment]
@dinki I just put on logging and searched and played some videos on beta 702 ..... I checked the log file and I also see this kind of errors - some type of errors in the logging do not mean you have trouble in viewing/listening to videos on sytv ; on the contrary, as the sytv-app is tweaked to avoid google-ad-stuff , it is normal that it generates "errors" in comparison with the normal working of the official yt-app.
I have no ads, so there is no correlation with these errors-type you suggest and so these are not the reason for your ads...

**Comment by dinki** [comment]
@foreveryoungforty So it seems a new beta came out yesterday so I upgraded.  The problem I mentioned above has gone away on my mibox but a new 'issue' has popped up.  Now when I push down to go past the UI controls and to the next/suggested area I get a black screen but still audio playing.  I press down again and I can see the next/suggested videos fine and can browse them but the video is still black.  I can exit and start the video again and it continues playing.

I'm not sure why I'm running into all these problems and others aren't but something is definitely changing for me.

**Comment by foreveryoungforty** [comment]
@dinki strange indeed I run 702 for 2 days now (updated 22/07) and no issues....

**Comment by dinki** [comment]
I guess I'm kind of talking to myself, but here goes.  I tested again on the latest beta and noticed that things were working just fine all of a sudden.  Tried another video and it was doing the same thing as before.  Seems that the video that had no problems was not monetized so it is definitely all videos that have ads in them that cause the app to have problems.  I've tried turning off the skip adverts option and ads do not play and exhibit the same problematic behavior.

For fun I decided to try on the Amazon Fire Stick and have the exact same problem.  I now have three separate devices that exhibit the exact same behavior.

**Comment by lostmsu** [comment]
Getting the same problem on Fire TV from U.S.

Can't get to app settings. Holding back button eventually shows "do you want to exit screen". Also, pause button on remote stopped working.

**Comment by rodalpho** [comment]
@yuliskov 

Adblocking is largely broken now. Running version 702. Same behavior in beta 707.

On Pro Main, pre-roll ads don't play, and mid-roll ads break the stream, causing a never-ending "loading" status circle. If there are no mid-roll ads, when you finish playing the video, it pops up the ad status indicator in the bottom right and you can hit the back button to get back to the video list, so it is possible to watch most videos without ads. However this breaks the ability to get to the channel, or hit the down button to see suggested videos. All you can do is exit. 

On Lite Main, ads play immediately. Not blocked at all.

**Comment by dinki** [comment]
Yes.  I've updated multiple times in the past few weeks and continue to see what you described above.  I am able to watch a single video, exit to video selection and choose another.  Cannot see the channel view without subscribing and picking it from the sub menu and cannot see/choose any of the suggested videos under the control gui element.

**Comment by rodalpho** [comment]
Indeed, it is _substantially_ broken now. Hope @yuliskov is OK, haven't seen any posts from him for 2 weeks here.

**Comment by rodalpho** [comment]
Seems to be fixed in beta 708! Thanks yuliskov!

**Comment by dinki** [comment]
Confirmed working for me as well!  Strange but I was not offered the new beta update but downloaded and sideloaded it and all working great.  Thanks for posting the news @rodalpho and for the fix @yuliskov !

**Comment by rodalpho** [comment]
The beta is a completely separate app now, with a blue icon. This makes sense because sometimes you'll get updates once per day. I plan to switch back to non-beta once the fix makes it there.

**Comment by ghzgod** [comment]
Can confirm I fully deleted the app and side loaded the beta and all ads are removed.  Thank you very much dev!
